src/jquery.interactive-world-map.js

```diff
@@ -1,142 +1,152 @@
-            }
-
-            if(margins && typeof(margins['marginCountryPopupForLeft']) != 'undefined') {
-                marginCountryPopupForLeft           = margins['marginCountryPopupForLeft'];
-            }
-
-            if(options['colors']) {
-                $.fn.interactiveWorldMap.defaults.highlighted  = options['colors']['highlighted'];
-                $.fn.interactiveWorldMap.defaults.hover        = options['colors']['hover'];
-            }
-
-        } else {
-
-            dataCountries = $(this).data('countries');
-            dataCountries = dataCountries['countries'];
+;$(function( $, window, document, undefined) {
+
+    var pluginName = "interactiveWorldMap";
+    var settings;
+    var dataCountries = {};                 //Hold the countries information sent by parameters
+    var countriesFill = {};                 //Hold the countries code that must be Highlighted for be a YFU Country
+    var hoverPattern;                       //The svg pattern (html code) that uses the hover colors over the YFU countries and make it dotted.
+    var yfuCountriesHighlighted;            //The pattern that makes the yfu countries dotted and highlighted
+    var generalDottedPatternStyle;          //The general dotted style for all the countries
+
+    var defaults = {
+        colors: {
+            general: '#cccccc',
+            hover: 'red',
+            highlighted: 'green'
+        },
+        popup: {
+            margins: {top: 100, left: 60}
+        },
+        svg: {
+           pattern: {
+                circle: {
+                    x: 1, y: 1, cx:3, cy: 3, r: 3
+                },
+               size: {
+                   width:  "5",
+                   heigth: "5"
+               }
+           }
+        }
+    };
 
-            $data = $(this).data();
 
-            if($data.margincountrypopupfortop) {
-                marginCountryPopupForTop            = $data.margincountrypopupfortop;
-            }
+    function interactiveWorldMap(element, options) {
+        this.element    = element;
+        this._options   = options;
+        this.init();
+    }
 
-            if($data.margincountrypopupforleft) {
-                marginCountryPopupForLeft          = $data.margincountrypopupforleft;
-            }
 
-            if($data.colorhover){
-                $.fn.interactiveWorldMap.defaults.hover       = $data.colorhover
-            }
-            if($data.colorhighlighted){
-                $.fn.interactiveWorldMap.defaults.highlighted = $data.colorhighlighted;
+    $.extend(interactiveWorldMap.prototype, {
+            init: function () {
+                settings   = $.extend(defaults, $(this.element).data(), this._options);
+                if ($.isPlainObject(settings.countries)) {
+                    dataCountries = settings.countries;
+                }else {
+                    dataCountries = jQuery.parseJSON(settings.countries);
+                }
```

Comment:
Maybe is there another better way? also I can create a function that checks the parameter and if it's string I do first a parseJSON and the go on oder...